src/main/java/com/iexec/common/tee/TeeSessionGenerationError.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2022 IEXEC BLOCKCHAIN TECH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.common.tee;
+
+public enum TeeSessionGenerationError {
```

Comment:
Can you remind me/explain why `TeeSessionGenerationError` isn't exposed by the sms-library itself?

Reply:
It is needed in `ReplicateStatusDetails` which is part of Common.

Reply:
Back to our usual debate - enum or String :)
But I think this would be acceptable there. Let's rework this!

Reply:
If the Enum works, I prefer we use it.
It adds type safety and more robustness in the system with a field taking value in a finite list of values.
Or did I misunderstand something ?

Reply:
I partially agree with you @jbern0rd. Having type safety is a great feature but having `TeeSessionGenerationError` in Common looks strange to me. I prefer it to be in the iexec-sms-library project.